src/main/java/org/kairosdb/core/datastore/KairosDatastore.java

```diff
@@ -248,9 +249,12 @@ public DatastoreQuery createQuery(QueryMetric metric) throws DatastoreException
 
 		DatastoreQuery dq;
 
+		Span span = GlobalTracer.get().activeSpan();
```

Comment:
Nitpick: why not use the @Inject'd tracer?  In practice they _should_ never be different, but better safe than sorry.  We're believe we are also using the @Inject'd tracer for all other tracing in this class as well, so it's good to be consistent.

Further nickpick: if this is one of the classes responsible for unit tests failing from tracer NPEs, this is a good opportunity to move that to the constructor for injection as well.